[gemonlinedb] Cannot build tests

<!--- Provide a general summary of the issue in the Title above -->
## Brief summary of issue
<!--- Provide a description of the issue, including any other issues or pull requests it references -->
PR #239 added an additional dependency on `gembase` and `gemutils`. This breaks `make tests`:
```
In Makefile.fallbacks: guessing/checking values
XDAQ_ROOT="/opt/xdaq"
XDAQ_OS already set to "linux"
XDAQ_PLATFORM already set to "x86_64_centos7"
Guessed undefined XDAQ_PLATFORM to "x86_64_centos7"
VER_EXISTS yes
PACKAGE_VER_MAJOR 0
PACKAGE_VER_MINOR 0
PACKAGE_VER_PATCH 0
LongPackageLoc GEMONLINEDB
BUILD_HOME /afs/cern.ch/user/l/lmoureau/GEM/gemonlinedb
BUILD_VERSION 1.0.138.dev
Trying to to determine flags for gcc(40805)/clang(40201)
Using GCC48Flags for GCC 40805
Calling make all in directory /afs/cern.ch/user/l/lmoureau/GEM/gemonlinedb/cmsgemos/gemonlinedb/.loop using tests
make -C /afs/cern.ch/user/l/lmoureau/GEM/gemonlinedb/cmsgemos/gemonlinedb/ -f Makefile _testsall
In Makefile.fallbacks: guessing/checking values
XDAQ_ROOT="/opt/xdaq"
XDAQ_OS already set to "linux"
XDAQ_PLATFORM already set to "x86_64_centos7"
Guessed undefined XDAQ_PLATFORM to "x86_64_centos7"
VER_EXISTS yes
PACKAGE_VER_MAJOR 0
PACKAGE_VER_MINOR 0
PACKAGE_VER_PATCH 0
LongPackageLoc GEMONLINEDB
BUILD_HOME /afs/cern.ch/user/l/lmoureau/GEM/gemonlinedb
BUILD_VERSION 1.0.138.dev
Trying to to determine flags for gcc(40805)/clang(40201)
Using GCC48Flags for GCC 40805
make[1]: Entering directory `/afs/cern.ch/user/l/lmoureau/GEM/gemonlinedb/cmsgemos/gemonlinedb'
mkdir -p /afs/cern.ch/user/l/lmoureau/GEM/gemonlinedb/cmsgemos/gemonlinedb/bin/linux/x86_64_centos7
g++   -o  /afs/cern.ch/user/l/lmoureau/GEM/gemonlinedb/cmsgemos/gemonlinedb/test/linux/x86_64_centos7/test/testAMC13Configuration.exe  /afs/cern.ch/user/l/lmoureau/GEM/gemonlinedb/cmsgemos/gemonlinedb/test/linux/x86_64_centos7/test/testAMC13Configuration.o  -L/afs/cern.ch/user/l/lmoureau/GEM/gemonlinedb/cmsgemos/gemonlinedb/lib/linux/x86_64_centos7 -L/opt/xdaq/lib -L/afs/cern.ch/user/l/lmoureau/GEM/gemonlinedb/x86_64_centos7/lib -L/opt/xdaq/lib \
-lgemonlinedb -lpthread -ldl -lgembase -lgemutils -lxdaq -lxdata -lmimetic -lxoap -lxerces-c -lxcept -ltoolbox -llog4cplus -lxoap -lboost_unit_test_framework -ltstoreapi -lxdata -lmimetic -lwsaddressing -ltstoreutils
/usr/bin/ld: cannot find -lgembase
/usr/bin/ld: cannot find -lgemutils
collect2: error: ld returned 1 exit status
make[1]: *** [/afs/cern.ch/user/l/lmoureau/GEM/gemonlinedb/cmsgemos/gemonlinedb/test/linux/x86_64_centos7/test/testAMC13Configuration.exe] Error 1
make[1]: Leaving directory `/afs/cern.ch/user/l/lmoureau/GEM/gemonlinedb/cmsgemos/gemonlinedb'
make: *** [/afs/cern.ch/user/l/lmoureau/GEM/gemonlinedb/cmsgemos/gemonlinedb/.loop] Error 2
```

### Types of issue
<!--- Propsed labels (see CONTRIBUTING.md) to help maintainers label your issue: -->
- [x] Bug report (report an issue with the code)
- [ ] Feature request (request for change which adds functionality)

## Expected Behavior
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->

`make tests` builds the tests and `make run-tests` runs them.

### Steps to Reproduce (for bugs)
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
1. Checkout `feature/gemonlinedb`
2. `make`
3. `cd gemonlinedb`
4. `make tests`

## Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

Using tests to check that I haven't broken anything while developping.

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version used: `feature/gemonlinedb`
* Shell used: `bash`

<!--- Template thanks to https://www.talater.com/open-source-templates/#/page/98 -->

--- a/Makefile
+++ b/Makefile
@@ -18,6 +18,7 @@ SUBPACKAGES.INSTALL  := $(patsubst %,%.install,  ${SUBPACKAGES})
 SUBPACKAGES.RPM      := $(patsubst %,%.rpm,      ${SUBPACKAGES})
 SUBPACKAGES.CLEANRPM := $(patsubst %,%.cleanrpm, ${SUBPACKAGES})
 SUBPACKAGES.CLEAN    := $(patsubst %,%.clean,    ${SUBPACKAGES})
+SUBPACKAGES.TESTS    := $(patsubst %,%.tests,    ${SUBPACKAGES})
 
 #OS:=linux
 #ARCH:=x86_64
@@ -81,6 +82,8 @@ cleanrpm: $(SUBPACKAGES.CLEANRPM)
 
 clean: $(SUBPACKAGES.CLEAN)
 
+tests: $(SUBPACKAGES) $(SUBPACKAGES.TESTS)
+
 $(LIBDIR):
 	mkdir -p $(LIBDIR)
 
@@ -98,6 +101,9 @@ $(SUBPACKAGES.INSTALL):
 
 $(SUBPACKAGES.CLEAN):
 	$(MAKE) -C $(patsubst %.clean,%, $@) clean
+
+$(SUBPACKAGES.TESTS):
+	$(MAKE) -C $(patsubst %.tests,%, $@) tests
 
 .PHONY: $(SUBPACKAGES) $(SUBPACKAGES.INSTALL) $(SUBPACKAGES.CLEAN)
 

--- a/gemonlinedb/Makefile
+++ b/gemonlinedb/Makefile
@@ -17,8 +17,6 @@ GEMONLINEDB_VER_PATCH=0
 include $(BUILD_HOME)/$(Project)/config/mfDefsGEM.mk
 include $(BUILD_HOME)/$(Project)/config/mfPythonDefsGEM.mk
 
-DependentLibraries += boost_filesystem
-
 Sources = \
     AMC13Configuration.cc \
     AMCConfiguration.cc \
@@ -41,8 +39,6 @@ Sources = \
 
 DynamicLibrary=gemonlinedb
 
-TestLibraries = $(DependentLibraries) xoap boost_unit_test_framework tstoreapi xdata mimetic wsaddressing tstoreutils
-
 TestExecutables = \
     test/testAMC13Configuration.cc \
     test/testConfigurationLinker.cc \
@@ -61,8 +57,13 @@ IncludeDirs+=$(BUILD_HOME)/$(Project)/gemutils/include
 DependentLibraryDirs+=$(BUILD_HOME)/$(Project)/gembase/lib/$(XDAQ_OS)/$(XDAQ_PLATFORM)
 DependentLibraryDirs+=$(BUILD_HOME)/$(Project)/gemutils/lib/$(XDAQ_OS)/$(XDAQ_PLATFORM)
 
-DependentLibraries=gembase gemutils xdaq xdata mimetic xoap xerces-c xcept toolbox log4cplus
+DependentLibraries=gembase gemutils xdaq cgicc xgi peer xdata mimetic xoap xerces-c xcept toolbox logudpappender logxmlappender log4cplus config cgicc
+DependentLibraries += boost_filesystem boost_regex boost_system
+
 # DependentLibraries+=$(StandardLibraries)
+
+TestLibraries= $(DependentLibraries) boost_unit_test_framework tstoreapi wsaddressing tstoreutils
+TestLibraryDirs= $(DependentLibraryDirs)
 
 # Targets with additional dependencies. Keep "default" first
 .PHONY: default clean tests

